src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -444,10 +444,11 @@ internal static CSDiagnosticInfo CheckAccessibility(DeclarationModifiers modifie
 
             if ((modifiers & DeclarationModifiers.Required) != 0)
             {
+                var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
```

Comment:
The property and containing types are all from source. They don't have use-site diagnostics to report.

Reply:
> The property/field and containing types are all from source. They don't have use-site diagnostics to report.

Digging through bases doesn't have to stay in source, etc. I think it will be less risky and more robust to simply report it.

Reply:
> Digging through bases

We don't dig through bases. We only look at containing types. The only things that can be considered here are source types.